### ISSUE TITLE
Split more pages into sub-topics

### ISSUE THREAD

**Issue by mattip** [pytorch/pytorch#38010]
## 📚 Documentation

gh-37419 split up the "heaviest" pages into a leading pages and subpages. More can be done to split up pages (these have the most lines of HTML):
- [ ] [`docs/source/quantization.rst](https://pytorch.org/docs/stable/quantization.html)
- [ ] [`docs/source/tensors.rst`](https://pytorch.org/docs/stable/tensors.html#torch.Tensor)
- [ ] [`docs/source/nn.fuctional.rst`](https://pytorch.org/docs/stable/nn.functional.html)
- [ ] [`docs/source/distributions.rst`](https://pytorch.org/docs/stable/distributions.html)
- [ ] [`docs/source/autograd.rst`](https://pytorch.org/docs/stable/autograd.html)

The strategy would start with using `autosummary` like in gh-37419, but more work might be needed to split up classes with many methods.

**Comment by glaringlee** [comment]
Add triage review label since assigning 'module:docs' won't cc POCs.
